server/src/main/java/org/candlepin/controller/SuspendModeTransitioner.java

```diff
@@ -132,4 +209,24 @@ private void cleanStatusCache() {
         candlepinCache.getStatusCache().clear();
     }
 
+    private void resetQpidReasons(Set<Reason> reasons, Reason ... keep) {
+        List<Reason> allAMQ = Arrays.asList(Reason.QPID_UP, Reason.QPID_DOWN, Reason.QPID_FLOW_STOPPED);
+        resetReasonGroup(reasons, allAMQ, keep);
+    }
+
+    private void resetActiveMQReasons(Set<Reason> reasons, Reason ... keep) {
+        List<Reason> allAMQ = Arrays.asList(Reason.ACTIVEMQ_DOWN, Reason.ACTIVEMQ_UP);
+        resetReasonGroup(reasons, allAMQ, keep);
+    }
+
+    private void resetReasonGroup(Set<Reason> reasons, List<Reason> reasonGroup, Reason ... keep) {
+        List<Reason> toKeep = keep != null ? Arrays.asList(keep) : new ArrayList<>();
```

Comment:
I feel like there's a better way to manage all this. I need some more time to digest it, but my gut is jumping to the streaming API.

Reply:
Will be addressed by ENT-982